cli/portforward.go

```diff
@@ -295,24 +308,72 @@ func parsePort(in string) (uint16, error) {
 	return uint16(port), nil
 }
 
-func parsePortPort(in string) (local uint16, remote uint16, err error) {
+type parsedSrcDestPort struct {
+	local, remote uint16
+}
```

Comment:
Why not just have the function return a portForwardSpec directly? You could pass it a listen and dial network and it would return `[]portForwardSpec` instead of having an intermediary type

Reply:
My thinking was probably influenced by the previous implementation, and keeping the type "plain until use". Why plain? For instance, if you see the test where we mangle the `portForwardSpec` to make testing more convenient, that's not ideal IMO. In fact I could see `parsePortForwards` returning `[]parsedSrcDestPort` if it was updated with `protocol string` field. Then all items could be post-processed into `portForwardSpec` as a last step `¯\_(ツ)_/¯`. That code would seem more re-usable if we ever want to parse the port format in other places.

I'm not married to this by any means, I can change it, but I don't see a clear benefit to either way of doing it.